src/main/java/com/puppycrawl/tools/checkstyle/XpathFileGeneratorAstFilter.java

```diff
@@ -0,0 +1,149 @@
+////////////////////////////////////////////////////////////////////////////////
+// checkstyle: Checks Java source code for adherence to a set of rules.
+// Copyright (C) 2001-2018 the original author or authors.
+//
+// This library is free software; you can redistribute it and/or
+// modify it under the terms of the GNU Lesser General Public
+// License as published by the Free Software Foundation; either
+// version 2.1 of the License, or (at your option) any later version.
+//
+// This library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+// Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public
+// License along with this library; if not, write to the Free Software
+// Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
+////////////////////////////////////////////////////////////////////////////////
+
+package com.puppycrawl.tools.checkstyle;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.stream.Collectors;
+
+import com.puppycrawl.tools.checkstyle.api.AuditEvent;
+import com.puppycrawl.tools.checkstyle.api.AutomaticBean;
+import com.puppycrawl.tools.checkstyle.api.DetailAST;
+import com.puppycrawl.tools.checkstyle.api.FileText;
+import com.puppycrawl.tools.checkstyle.api.LocalizedMessage;
+import com.puppycrawl.tools.checkstyle.xpath.XpathQueryGenerator;
+
+/**
+ * Catches {@code TreeWalkerAuditEvent} and generates corresponding xpath query.
+ * Stores event and query inside static variables for {@code XpathFileGeneratorAuditListener}.
+ * See issue #102 https://github.com/checkstyle/checkstyle/issues/102
+ *
+ */
+public class XpathFileGeneratorAstFilter extends AutomaticBean implements TreeWalkerFilter {
+
+    /** The delimiter between xpath queries. */
+    private static final String DELIMITER = " | \n";
+
+    /** Default distance between tab stops. */
+    private static final int DEFAULT_TAB_WIDTH = 8;
+
+    /** List of the {@code TreeWalkerAuditEvent} objects, which supports xpath suppressions. */
+    private static final List<TreeWalkerAuditEvent> EVENTS = new ArrayList<>();
+
+    /** List of xpath queries for corresponding {@code TreeWalkerAuditEvent} objects. */
+    private static final List<String> QUERIES = new ArrayList<>();
+
+    /** Array of the checks, which supports xpath queries. */
+    private static final String[] SUPPORTED_CHECKS = {
+        "CyclomaticComplexity", "DeclarationOrder", "DefaultComesLast",
+        "ExplicitInitialization", "FallThrough", "HiddenField", "IllegalThrows",
```

Comment:
This list isn't needed and will be a pain to maintain while we work on support. It is already a pain to update list of checks supported in xdoc.
Since we are dealing with `TreeWalkerAuditEvent`, all you need to check is `event.getLocalizedMessage().getColumnCharIndex() != 0` or `event.getLocalizedMessage().getTokenType() != 0`.

Reply:
But if I am not wrong `getColumnCharIndex()` can be zero 
Probably we can use `getTokenType()!=0`

Reply:
Yes, use tokenType. If it is still an issue, we will need to re-evaluate our default values in `LocalizedMessage` when we aren't given a value for it.

Reply:
done